src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -132,7 +133,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
                     $this->hardCopy($originDir, $targetDir);
                 }
             }
+            if (is_dir($targetDir)) {
+                array_push($validAssetDir, $targetDir);
+            }
+        }
+        // Check in $bundlesDir, if all links/folder still have an existing Bundle
+        if ($dir = opendir($bundlesDir)) {
```

Comment:
use a `DirectoryIterator`. The code will be simpler:

- no need to skip `.` and `..` (you can configure the iterator to skip them for you)
- no need to close the directory at the end (btw, you don't properly close it if `$filesystem->remove` throws an exception)
- you can just do a `foreach`